### DISCUSSION TITLE
Updating dependencies

### DISCUSSION THREAD

**Discussion by winksaville** [schteve/nucleo-f446re#5] [General]
I've started updating my WIP branch, https://github.com/winksaville/nucleo-f446re/commits/WIP-Upgrade-dependencies-to-latest-versions/ , I've got 3 commits  1st updated the deps. The second I commented out LedAnalog because it wouldn't compile and modified blinky_fade_direct and got it to compile but it doesn't work. So in the 3rd commit I added rprintln.rs to see if I could get printf debugging working. It does and the LED blinks, so a little progress. I'm planning to use printf debugging to see if I can get fade and then LedAnalog to work. I'm going to try to work on it some more tomorrow, but your welcome to use anything I've done as u see fit.

**Comment by winksaville** [comment]
Oh, I used stm32-rs/stm32f4xx-hal/blob/master/examples for tweaking blinky_fade_direct.

**Comment by schteve** [comment]
Think I got it.

- The only embedded HAL change that affected things was the unification of the delay interfaces. Importing DelayNs instead of DelayMs was enough to make that work.
- The STM32F4xx crate change was easier, as you noted. They make you construct a channel explicitly rather than deducing it from the pin.
- Other crate changes should be transparent.
